src/fitnesse/slim/SlimExpressionEvaluator.java

```diff
@@ -14,21 +14,13 @@
  * Evaluates slim expressions.
  */
 public class SlimExpressionEvaluator {
-  private static final ScriptEngineManager ENGINE_MANAGER = new ScriptEngineManager();
 
-  private ScriptEngine engine;
+  private final ScriptEngine engine;
 
   public SlimExpressionEvaluator() {
-    this("JavaScript");
+    this.engine = new NashornScriptEngineFactory().getScriptEngine();
   }
 
-  public SlimExpressionEvaluator(String engineName) {
-    this(ENGINE_MANAGER.getEngineByName(engineName));
-  }
-
-  public SlimExpressionEvaluator(ScriptEngine engine) {
```

Comment:
Why not keep this constructor and feed it the nashhorn engine?

Reply:
I thought to do it this way, so we had the nashorn import and scriptengine initialization only in this file. I can see the compatibility argument to keep the old constructor. Not really a preference for one or the other.